# CAVE
Cancer Associated variant enrichment

This repository contains the code and data neede to reproduce the main findings in the study:

**Cancer Associated Variant Enrichment (CAVE), a gene-agnostic approach to identify low burden variants; application to TP53 and chronic lymphocytic leukemia**


## Repository Contents

This repository contains the following:

- **Code**: Scripts and R Markdown (.Rmd) files to reproduce the analyses.
- **Data**: Processed data and raw VCF data of mutations.

## Instructions for Running the Code

To reproduce the analyses, follow the steps outlined in the provided .Rmd files. These files contain the necessary code and instructions.

## Dependencies and System Requirements

Ensure you have the following dependencies installed:

- R (version 3.6.0 or higher)
- R packages: dplyr, tidyr, stringr, vcfR, ggplot2, readxl, MutSigTools, MutationalPatterns, reshape2, dNdScv. 

## Usage

1. Clone the repository:
    ```sh
    git clone https://github.com/yourusername/CAVE.git
    cd CAVE
    ```

2. Open the .Rmd files in RStudio or another R environment.

3. Follow the instructions in the .Rmd files to run the analyses.

## License

This code and data are licensed for academic use only.



